### PR TITLE
Use cornice swagger to generate the OpenAPI documentation

### PR DESCRIPTION
**Pull request by gabisurita**
Fixes: #1006 and #707
Replaces: #1033

- [x] Add security information.
- [x] Fix/add tests.
- [x] Add documentation.
- [x] Cleanup YAML files.

Next steps (which I don't plan to do on this PR):
- Validate ids (related to #1077)
- Add single resource documentation in options (#707)
- Use pyramid_swagger for testing
- Improve schema details/descriptions

### CODE REVIEW

File: kinto/core/resource/schema.py [hunk 1]
```diff
@@ -331,3 +332,130 @@ class JsonPatchRequestSchema(RequestSchema):
...
+
+        return bound
+
+
+class ShareableResourseResponses(ResourceReponses):
+    """Class that wraps and handles SharableResource responses."""
+
+    def __init__(self):
+        super(ShareableResourseResponses, self).__init__()
+        self.default_schemas.update({
```
Comment (glasserc) [inline kinto/core/resource/schema.py]: I think this will mutate the shared `default_schemas` dictionary.

File: kinto/core/api.py [hunk 2]
```diff
@@ -69,3 +106,19 @@ def default_op_ids(self, service, method):
...
+    def default_security(self, service, method):
+        """Provides OpenAPI security properties based on kinto policies."""
+
+        definitions = service.definitions
+
+        # Get method view arguments
+        for definition in definitions:
+            met, view, args = definition
+            if met == method:
+                break
```
Comment (gabisurita) [inline kinto/core/api.py]: This is never supposed to happen. The ``method`` that is passed to the ``default_security`` is actually fetched the same way that is done here (by inspecting service definitions), but the interface for default_xxx calls on cornice swagger is designed to take only ``service`` as a parameters (so we can have the same interface for all defaults), but we can change the api to take a method definition as an optional argument.

**Comment by gabisurita** [comment]
Final r? @Natim @leplatrem

File: kinto/core/views/batch.py [hunk 3]
```diff
@@ -67,13 +68,43 @@ class BatchRequest(colander.MappingSchema):
...
+
 batch = Service(name="batch", path='/batch',
                 description="Batch operations")
 
 
 @batch.post(schema=BatchRequest,
             validators=(colander_validator,),
-            permission=NO_PERMISSION_REQUIRED)
+            permission=NO_PERMISSION_REQUIRED,
+            tags=['Batch'], operation_id='batch',
```
Comment (leplatrem) [inline kinto/core/views/batch.py]: nit: shouldn't these go to the service definition ?

Comment (gabisurita) [inline kinto/core/views/batch.py]: Tags could be defined on the service, but cornice swagger doesn't support it yet (There's an issue for that https://github.com/Cornices/cornice.ext.swagger/issues/61). Operation ids need to be isolated by method/verb.

**Pull request merged**